projects/igniteui-angular/src/lib/grids/column.component.ts

```diff
@@ -1025,6 +1042,54 @@ export class IgxColumnComponent implements AfterContentInit {
             }
         }
     }
+
+    /**
+     * @hidden
+     */
+    getGridTemplate(isRow, isIE) {
+        const itemAccum = isRow ?
+            (acc, val) => Math.max(val.rowStart + val.gridRowSpan - 1, acc) :
+            (acc, val) => Math.max(val.colStart + val.gridColumnSpan - 1, acc);
+        const templateItems = this.children && this.children.reduce(itemAccum, 1) || 1;
+        const generatedSizes = !isRow ? this.generateColumnSizes(this.children) : null;
+        return isIE ?
+        generatedSizes || `(1fr)[${templateItems}]` :
+            generatedSizes || `repeat(${templateItems},1fr)`;
+    }
+
+    protected generateColumnSizes(children) {
+        const columnSizes = [];
+        // find the smallest col spans
+        children.forEach(col => {
+            const actualWidth = parseFloat(col.calcWidth) + 'px';
+            if (col.colStart && columnSizes[col.colStart - 1] === undefined) {
+                columnSizes[col.colStart - 1] = { colSpan: col.gridColumnSpan,
+                     width: actualWidth, widthSetByUser: col.widthSetByUser };
+            } else if (col.colStart && columnSizes[col.colStart - 1].colSpan > col.gridColumnSpan &&
+                (col.widthSetByUser || !columnSizes[col.colStart - 1].widthSetByUser)) {
+                columnSizes[col.colStart - 1] = { colSpan: col.gridColumnSpan,
+                    width: actualWidth, widthSetByUser: col.widthSetByUser };
+            }
+        });
+
+        // fill the gaps if there are any
+        let result = '';
+        for (let i = 0; i < columnSizes.length; i++) {
+            if (columnSizes[i] && columnSizes[i].colSpan !== 1) {
+                for (let j = 0; j < columnSizes[i].colSpan; j++) {
+                    result += ' ' +
+                    (columnSizes[i].widthSetByUser ?
+                        parseFloat(columnSizes[i].width) / columnSizes[i].colSpan + 'px' :
+                        columnSizes[i].width);
+                }
+                i += columnSizes[i].colSpan - 1;
+            } else if (columnSizes[i]) {
+                result += ' ' + columnSizes[i].width;
+            }
+        }
+        return result.trim();
```

Comment:
at a glance something called generateColumnSize**s** which returns a single string is very strange.

please, refactor